src/Nix/Options/Parser.hs

```diff
@@ -27,140 +29,163 @@ nixOptions :: UTCTime -> Parser Options
-          )
+        (  long "check"
+        <> help "Whether to check for syntax errors after parsing"
+        )
     <*> optional
-          (strOption
-            (  long "read"
-            <> help "Read in an expression tree from a binary cache"
-            )
+        (strOption
+          (  long "read"
+          <> help "Read in an expression tree from a binary cache"
           )
+        )
     <*> switch
-          (  long "cache"
-          <> help "Write out the parsed expression tree to a binary cache"
-          )
+        (  long "cache"
+        <> help "Write out the parsed expression tree to a binary cache"
+        )
     <*> switch
-          (  long "repl"
-          <> help "After performing any indicated actions, enter the REPL"
-          )
+        (  long "repl"
+        <> help "After performing any indicated actions, enter the REPL"
+        )
     <*> switch
-          (  long "ignore-errors"
-          <> help "Continue parsing files, even if there are errors"
-          )
+        (  long "ignore-errors"
+        <> help "Continue parsing files, even if there are errors"
+        )
     <*> optional
-          (strOption
-            (short 'E' <> long "expr" <> help "Expression to parse or evaluate")
-          )
+        (strOption
+          (  short 'E'
+          <> long "expr"
+          <> help "Expression to parse or evaluate")
+        )
     <*> many
-          (argPair
-            (long "arg" <> help "Argument to pass to an evaluated lambda")
-          )
+        (argPair
+          (  long "arg"
+          <> help "Argument to pass to an evaluated lambda")
+        )
     <*> many
-          (argPair
-            (  long "argstr"
-            <> help "Argument string to pass to an evaluated lambda"
-            )
+        (argPair
+          (  long "argstr"
+          <> help "Argument string to pass to an evaluated lambda"
           )
+        )
     <*> optional
-          (strOption
-            (short 'f' <> long "file" <> help
-              "Parse all of the files given in FILE; - means stdin"
-            )
+        (strOption
+          (  short 'f'
+          <> long "file"
+          <> help "Parse all of the files given in FILE; - means stdin"
           )
+        )
     <*> option
-          (parseTimeOrError True defaultTimeLocale "%Y/%m/%d %H:%M:%S" <$> str)
-          (long "now" <> value current <> help
-            "Set current time for testing purposes"
+        (parseTimeOrError True defaultTimeLocale "%Y/%m/%d %H:%M:%S" <$> str)
+        (  long "now"
+        <> value current
+        <> help "Set current time for testing purposes"
+        )
+    <*> many
+        (strArgument
+          (  metavar "FILE"
+          <> help "Path of file to parse"
           )
-    <*> many (strArgument (metavar "FILE" <> help "Path of file to parse"))
+        )
+
+versionOpt :: Parser (a -> a)
+versionOpt = infoOption
+             (showVersion version)
```

Comment:
I usually prefer arguments to be indented relatively to the function, but I don't care too much.

Reply:
Yep.

You right.